k8s/solr/README.md

```diff
@@ -0,0 +1,417 @@
+community defines the Application resource. You can find the source code at
+[github.com/kubernetes-sigs/application](https://github.com/kubernetes-sigs/application).
+
+### Installing the app
+
+Navigate to the `solr` directory:
+
+```shell
+cd click-to-deploy/k8s/solr
+```
+
+#### Configuring the app with environment variables
+
+Choose the instance name and namespace for the app. For most cases, you can
+use the `default` namespace.
+
+```shell
+export APP_INSTANCE_NAME=solr-1
+export NAMESPACE=default
+```
+(Optional) Enable Stackdriver Metrics Exporter:
+
+> **NOTE:** Your GCP project must have Stackdriver enabled. If you are using a
+> non-GCP cluster, you cannot export metrics to Stackdriver.
+
+By default, the application does not export metrics to Stackdriver. To enable
+this option, change the value to `true`.
+
+```shell
+export METRICS_EXPORTER_ENABLED=false
+```
+
+Configure the image tag:
+
+```shell
+export TAG=8.6
+export ZK_TAG=3.6
+```
+Configure the container images:
+
+```shell
+export IMAGE_SOLR="marketplace.gcr.io/google/solr8"
+export IMAGE_ZOOKEEPER="marketplace.gcr.io/google/zookeeper3"
+export IMAGE_DEPLOYER="marketplace.gcr.io/google/solr/deployer:${TAG}"
+```
+
+Set or generate the passwords:
+
+```shell
+# SolrCloud UI console
+export SOLR_USER="solr"
+export SOLR_PASSWORD=$(cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 10 | head -n 1 | tr -d '\n')
+
+# ZooKeeper ACL credentials
+export ZK_READONLY_USER="readonly-user"
+export ZK_READONLY_PASSWORD=$(cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 10 | head -n 1 | tr -d '\n')
+
+export ZK_ADMIN_USER="admin-user"
+export ZK_ADMIN_PASSWORD=$(cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 10 | head -n 1 | tr -d '\n')
+```
+
+Set the storage class for the persistent volume of SolrCloud nodes and ZooKeeper nodes:
+
+ * Set the StorageClass name. You can select your existing StorageClass name for
+   the persistent disk of SolrCloud application storage.
+ * Set the persistent disk's size for SolrCloud storage. The default disk size is
+   `10Gi`.
+ * Set the persistent disk's size for ZooKeeper storage. The default disk size
+   is `5Gi`.
+
+```shell
+export STORAGE_CLASS="standard" # provide your StorageClass name if not "standard"
+export PERSISTENT_SOLR_SIZE="10Gi"
+export PERSISTENT_ZK_SIZE="5Gi"
+```
+#### Creating a namespace in your Kubernetes cluster
+
+If you use a different namespace than `default`, or if the namespace does
+not exist yet, create a new namespace:
+
+```shell
+kubectl create namespace "${NAMESPACE}"
+```
+
+#### Expanding the manifest template
+
+To expand the template, use `helm template`. We recommend that you save the
+expanded manifest file for future updates to your app.
+
+```shell
+helm template "${APP_INSTANCE_NAME}" \
+  --namespace "${NAMESPACE}" \
+  --set solr.image.repo="${IMAGE_SOLR}" \
+  --set solr.image.tag="${TAG}" \
+  --set zookeeper.image="${IMAGE_ZOOKEEPER}:${ZK_TAG}" \
```

Comment:
Here it's something that IMO it's missing consistency when we compare: `zookeeper.image` and `deployer.image`. 

IMAGE_DEPLOYER receive the full path and IMAGE_ZOOKEEPER doesn't. I'd keep consistent.

Reply:
IMHO it's okay, only main image is using image and tag, all the rest are full images string